File: Introductory_Problems/Java/Repetitions.java
package CSES_Solved_Questions.Introductory_Problems.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;

public class Repetitions {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter writer = new PrintWriter(System.out);
        String s = reader.readLine();
        char prev = '#';

        int len = 1, ans = 1;
        for(char c: s.toCharArray()){
            if(prev == c){
                len++;
            }
            else{
                ans = Math.max(ans, len);
                len = 1;
                prev = c;
            }
        }

        ans = Math.max(ans, len);

        writer.println(ans);

        reader.close();
        writer.close();
    }
}

File: Introductory_Problems/C++/Repetitions.cpp
#include <iostream>

using namespace std;

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    string seq;
    cin>>seq;

    int len = 1;
    int ans = 1;

    
    char prev = '#';
    for (auto c : seq){
        if(prev == c){
            len++;
        }
        else{
            ans = max(ans, len);
            len = 1;
            prev = c;
        }
    }
    ans = max(ans, len);
    cout<<ans;
}